lib/bolt/applicator.rb

```diff
@@ -132,6 +132,7 @@ def compile(target, ast, plan_vars)
     def validate_hiera_config(hiera_config)
       if File.exist?(File.path(hiera_config))
         data = File.open(File.path(hiera_config), "r:UTF-8") { |f| YAML.safe_load(f.read, [Symbol]) }
+        raise Bolt::ParseError, "No hiera data found in #{hiera_config}" unless data
```

Comment:
We allow an empty `bolt.yaml` and `inventory.yaml`. I see the `hiera.yaml` as a similar config file which I think we should allow to be empty. maybe we should just handle that case gracefully? Also If we decide to keep the error, I think hiera config is separate from hiera data (hiera config tells puppet/bolt how to look up hiera data), so saying `no hiera data found` should probably say `invalid hiera configuration`. But again, I think that empty should be valid.